Subject: fix annotations of xen-head.S
From: Jiri Slaby <jslaby@suse.cz>
Patch-mainline: submitted on 12/9/2017
References: bnc#1058115

On i386, arch/x86/kernel/head_32.S includes xen-head.S too. Since it is
annotated now, the assembler refuses to compile the file:
arch/x86/kernel/../../x86/xen/xen-head.S:47: Error: no such instruction: `unwind_hint_empty'
scripts/Makefile.build:413: recipe for target 'arch/x86/kernel/head_32.o' failed

Include ../entry/calling.h to pull asm/unwind_hints.h where these macros are
defined.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/kernel/head_32.S |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/kernel/head_32.S
+++ b/arch/x86/kernel/head_32.S
@@ -25,6 +25,7 @@
 #include <asm/bootparam.h>
 #include <asm/export.h>
 #include <asm/pgtable_32.h>
+#include "../entry/calling.h"
 
 /* Physical address */
 #define pa(X) ((X) - __PAGE_OFFSET)
